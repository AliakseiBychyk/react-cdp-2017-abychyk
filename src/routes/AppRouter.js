import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import HomePage from '../components/HomePage/HomePage'
import SearchResultPage from '../components/SearchResultPage/SearchResultPage'
import MovieDetailPage from '../components/MovieDetailPage/MovieDetailPage'
import NotFoundPage from '../components/NotFoundPage/NotFoundPage'

const AppRouter = () => (
  <Router>
    <Switch>  
      <Route exact path="/" component={HomePage} />
      <Route path="/search/:searchQuery" component={SearchResultPage} />
      <Route path="/film/:show_title" component={MovieDetailPage} />
      <Route component={NotFoundPage} />
    </Switch>  
  </Router>
)

export default AppRouter