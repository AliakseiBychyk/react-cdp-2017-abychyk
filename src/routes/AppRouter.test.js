import React from 'react'
import { shallow } from 'enzyme'
import AppRouter from './AppRouter'

test('should render AppRouter correctly', () => {
  const wrapper = shallow(<AppRouter />)
  expect(wrapper).toMatchSnapshot()
})