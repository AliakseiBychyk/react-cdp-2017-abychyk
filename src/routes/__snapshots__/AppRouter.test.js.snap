// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppRouter correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/search/:searchQuery"
    />
    <Route
      component={[Function]}
      path="/movie/:show_title"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
